### DISCUSSION TITLE
🚀 FR Add check for form post integrity re: max_input_vars

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#4225] [Feature Requests]
### Suggest an idea for ExpressionEngine

This comes up periodically, and it came up twice this week.  Grids/Fluid fields can use a LOT fields, resulting in a lot of post variables.  If it hits the default max_input_vars, it drops the excess.  This either leads to errors or actual loss of data.

<img width="1143" alt="Screenshot 2024-04-12 at 9 29 27 AM" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/1181219/0921c859-3199-4841-8480-eed3f72ac401">

We upped max_input_vars and it fixed it.  The other one is unconfirmed, but they were saving and losing some grid rows.  Another case, I think they were getting a csrf error because the token was getting dropped.

I'd love it if at the end of every form, we have a 'check' variable that we do nothing but check for the presence.  If it's not there, the data hitting the server can't be considered 'intact' and it should show a validation warning instead.  Or we can up the max vars or some such.  

It doesn't come up a lot, but it comes up and can have weird, unpredictable effects.  It's one of the first things I check for when forms are getting weird results.

**Comment by intoeetive** [comment]
@robinsowell please check https://github.com/ExpressionEngine/ExpressionEngine/pull/4247

It's adding a hard stop when max_input_vars is too low for request. It's not looking nice - because there's a warning thrown by PHP (that we can't hide because the warning is thrown before our error handler is registered - unless @bryannielsen finds a way) and an error shown by EE, but it's preventing the data loss
